### PR TITLE
🛠  Add Pyolite Support

### PR DESCRIPTION
**Pull request by stevejpurves**
Aim is to add support for the pyolite kernel by launching the JupyterLiteServer in browser and connecting thebe to that.

### Status

* JupyterLiteServer is starting with required kernel extensions

### TODO

* need to figure out how to setup thebe to connect to it in both prod, and in dev with webpack dev server

### CODE REVIEW

**Comment by jtpio** [comment]
This looks promising, thanks @stevejpurves for starting this!

File: src/jlite.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  // Add the base serverlite extensions
+  const baseServerExtensions = await Promise.all(serverExtensions);
+  baseServerExtensions.forEach((p) => {
+    for (let plugin of activePlugins(p)) {
+      litePluginsToRegister.push(plugin);
+    }
+  });
+
+  // create the in-browser JupyterLite Server
+  const jupyterLiteServer = new JupyterLiteServer({} as any);
```
Comment (stevejpurves) [inline src/jlite.ts:40]:
```
const   {serviceManager} = juptyerLiteServer;
...
await jupyterLiteServer.start();
...
return serviceManager.serverSettings

```

Comment (stevejpurves) [reply src/jlite.ts:40]: server settings can be given to a thebe client, in order to connect to the server

**Comment by mmcky** [comment]
thank you @stevejpurves for working on this. I don't have much `javascript` talents but I'd be more than happy to invest time to get a `beta` up and running for one of our "lecture" site projects such as: https://python-programming.quantecon.org/intro.html it would be amazing to have live interactivity while learning python.

**Comment by stevejpurves** [comment]
definitely, @mmcky let's do that! in your book it doesn't look like thebe accessible at the moment, correct? I can see I loaded on page, but couldn't see UI to make it live.

**Comment by mmcky** [comment]
> definitely, @mmcky let's do that! in your book it doesn't look like thebe accessible at the moment, correct? I can see I loaded on page, but couldn't see UI to make it live.

@stevejpurves thanks! Currently I have just played with `sphinx-jupyterlite` https://github.com/QuantEcon/lecture-python-programming.myst/pull/181 to see how it works with `sphinx`. 

Currently the site plugs into `mybinder` via the theme link (top-right) but I'll go ahead and setup a branch to enable `thebe` on the `code-cells` directly. https://sphinx-thebe.readthedocs.io/en/latest/

**Update:** @stevejpurves I opened a branch to setup `thebe` but it looks like our publishing context (with notebooks in a separate repo to the source files) doesn't quite fit the `thebe` setup in `sphinx_book_theme` for connecting to the notebook version of an `md` source file (https://github.com/QuantEcon/lecture-python-programming.myst/pull/182). We might update `quantecon_book_theme` to modify the `sphinx_book_theme` setup slightly so the `live` button can connect to our `.notebooks` repository (that supports our live site and `mybinder`). cc @AakashGfude 

**Update 2:** I was able to get the setup working with `jupytext` and a `postBuild` file on the source `md` files. So thebe (powered by `mybinder`) is setup now on https://github.com/QuantEcon/lecture-python-programming.myst/pull/182 with demo: https://62877fcfe2ac9c06d1c747dd--epic-agnesi-957267.netlify.app/. The actual `launch `mybinder` link serves up the `md` file but the in page `thebe` cells work 🎉 

**Update 3:** I have `thebe` setup as per pull request above. Finding the connection to `mybinder` is triggering `rebuilds` instead of using a cached version -- so not sure what `config` I have missed on the my binder side. But the connections are in place. @stevejpurves let me know if you'd like me to do any work to test `pyolite` when you're ready. https://github.com/QuantEcon/lecture-python-programming.myst/pull/182#issuecomment-1136808471

**Comment by mmcky** [comment]
@stevejpurves so it looks like `mybinder` had some issues this past week with a couple of the federation members. So the issues I was facing were to do with `mybinder` rather than my config. So I think it's all setup now. It works nicely today with `mybinder` and launching. Let me know how I can help test this branch for you when you're ready.

**Pull request merged**